src/app/(tools)/square-image/square-tool.tsx

```diff
@@ -1,15 +1,41 @@
 "use client";
 
-import React, { useState, useEffect, type ChangeEvent } from "react";
+import React, { useState, useEffect, type ChangeEvent, useRef} from "react";
 import { usePlausible } from "next-plausible";
 import { useLocalStorage } from "@/hooks/use-local-storage";
 
+const ColorPicker = ({
+  onChange,
+  value,
+}: {
+  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
+  value: string;
+}) => {
+  const refInput = useRef<HTMLInputElement>(null);
+
+  return (
+    <>
+      <button
+        onClick={() => refInput.current?.click()}
```

Comment:
The common way to do this is usually to use a label, not to wrap the input in a button

Reply:
so I tried that.

so the color picker in already inside of a label, if I move the input, outside of the button it, will take space 

<img width="290" alt="Screenshot 2024-11-10 at 10 49 18 AM" src="https://github.com/user-attachments/assets/d92f2bf2-32a6-4958-a980-a863c1dbe5d5">


```tsx
      <button
        onClick={() => refInput.current?.click()}
        className="size-8 shrink-0 rounded-full border border-white"
        style={{ backgroundColor: value }}
      >

      </button>
      {/** issue: input can't be styled */}
      {/** https://stackoverflow.com/questions/48832432/rounded-input-type-color */}
      <input
        ref={refInput}
        type="color"
        className="invisible"
        value={value}
        onChange={onChange}
      />
      ```
      
      if I change the className to 'hidden', the color picker will show up at the top of the page.


other solution is to add w-0 but the color picker seems it looks a little off, 
<img width="453" alt="Screenshot 2024-11-10 at 10 51 56 AM" src="https://github.com/user-attachments/assets/0beacd60-be65-4359-9773-fd4c9164d5eb">
<img width="376" alt="Screenshot 2024-11-10 at 10 52 40 AM" src="https://github.com/user-attachments/assets/7c41a1b6-cbda-4cdc-ba00-44cb4b3b5806">

so that's why I think, having it inside of the button is better

Reply:
this one, seem close enough, classNam''
<img width="376" alt="Screenshot 2024-11-10 at 10 52 40 AM" src="https://github.com/user-attachments/assets/7f3e241b-f39a-4708-837b-18c121287188">

```tsx
      {/** issue: input can't be styled */}
      {/** https://stackoverflow.com/questions/48832432/rounded-input-type-color */}
      <input
        ref={refInput}
        type="color"
        className="invisible w-0"
        value={value}
        onChange={onChange}
      />
      <button
        onClick={() => refInput.current?.click()}
        className="size-8 shrink-0 rounded-full border border-white"
        style={{ backgroundColor: value }}
      >

      </button>
```

I will do this

Reply:
added a new commit